dpnp/backend/kernels/dpnp_krnl_arraycreation.cpp

```diff
@@ -127,6 +127,38 @@ void dpnp_ones_like_c(void* result, size_t size)
     dpnp_ones_c<_DataType>(result, size);
 }
 
+template <typename _DataType_input, typename _DataType_output>
+void dpnp_vander_c(void* array1_in, void* result1, size_t size_in, size_t N, int increasing)
+{
+    if (!size_in or !N) return;
+    _DataType_input* array_in = reinterpret_cast<_DataType_input*>(array1_in);
+    _DataType_output* result = reinterpret_cast<_DataType_output*>(result1);
+
+    if (N == 1) {
+        for (size_t i = 0; i < size_in; ++i) { result[i] = 1; }
+        return;
+    }
+
+    if (increasing) {
+        for (size_t i = 0; i < size_in; ++i) { result[i * N] = 1; }
+        for (size_t i = 1; i < N; ++i){
+            for (size_t j = 0; j < size_in; ++j){
+                result[j * N + i] = result[j * N + i - 1] * array_in[j];
+            }
+        }
+
+    } else {
+        int N_int = static_cast<int>(N);
+        int size_in_int = static_cast<int>(size_in);
```

Comment:
Are you sure, that you need this casts?

Reply:
I would use `size_t` instead of `int` in loops below and avoid using extra variables.